client/src/main/scala/grasshopper/client/ServiceClient.scala

```diff
@@ -3,26 +3,46 @@ package grasshopper.client
 import akka.actor.ActorSystem
 import akka.http.scaladsl.Http
 import akka.http.scaladsl.model.HttpMethods._
-import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
-import akka.stream.ActorFlowMaterializer
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model._
 import akka.stream.scaladsl.{ Sink, Source }
+import akka.stream.{ ActorFlowMaterializer, StreamTcpException }
 import akka.util.Timeout
 import com.typesafe.config.Config
-
-import scala.concurrent.Future
+import grasshopper.client.model.ResponseError
+import grasshopper.client.protocol.ClientJsonProtocol
 import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
 
-trait ServiceClient {
+trait ServiceClient extends ClientJsonProtocol {
   implicit val askTimeout: Timeout = 1000.millis
   implicit val system: ActorSystem = ActorSystem("grasshopper-client")
   implicit val materializer: ActorFlowMaterializer = ActorFlowMaterializer()
 
+  def host: String
+  def port: String
+
   val config: Config
 
-  def sendGetRequest(host: String, port: Int, path: String): Future[HttpResponse] = {
-    val connectionFlow = Http().outgoingConnection(host, port)
+  def sendGetRequest(path: String): Future[HttpResponse] = {
+    implicit val ec: ExecutionContext = system.dispatcher
+    val connectionFlow = Http().outgoingConnection(host, port.toInt)
     val request = HttpRequest(GET, path)
-    Source.single(request).via(connectionFlow).runWith(Sink.head)
+    Source.single(request).via(connectionFlow).runWith(Sink.head).recover {
+      case e: StreamTcpException =>
+        HttpResponse(
+          ServiceUnavailable,
+          Nil,
+          HttpEntity.empty(ContentTypes.NoContentType),
+          HttpProtocols.`HTTP/1.1`
+        )
+    }
+  }
+
+  def sendResponseError(response: HttpResponse) = {
+    Future.successful {
```

Comment:
Should we log the error response here, or is that already taken care of elsewhere by Akka?  Before you were including `${entity}` in the error message.  Probably useful to log that as well?  Just want to make sure we know when things are behaving badly.